### PR TITLE
obs-qsv11: Fix VPL initialization on intel-mediaSDK

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

VPL does not have a consistent initialization process for the various
backends it dispatches to. So this resolves the new VPL dispatcher
failing to initialize the video session when it dispatches to
intel-mediaSDK backend. On the oneVPL-intel-gpu backend VA-API sessions
are correctly initialized so this wasnt noticed when tested on newer
hardware.

I also tried to sneak in a small fixup for the low power fallback inside `InitParams`.
we call Query internally and possibly fixup the low power state and expect the
next status check to be done after the 2nd Query call.

### Motivation and Context

QSV Should work on 7th gen like it did before the shift to the vpl dispatcher.

### How Has This Been Tested?

Arc card and 7th gen chip both work again.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-qsv11/CMakeLists.txt [hunk 1]
```diff
@@ -58,5 +58,5 @@ elseif(OS_LINUX)
 
   target_sources(obs-qsv11 PRIVATE common_utils_linux.cpp)
 
-  target_link_libraries(obs-qsv11 PRIVATE Libva::va Libva::drm)
+  target_link_libraries(obs-qsv11 PRIVATE Libva::va Libva::drm Libva::wayland)
```
Comment (tytan652) [inline plugins/obs-qsv11/CMakeLists.txt:61]: Libva has it's own X11 library like DRM and Wayland.
To avoid linkage issue it should be added too.

Comment (kkartaltepe) [reply plugins/obs-qsv11/CMakeLists.txt:61]: It doesnt provide any actual linkages, vaGetDisplay is X11 by definition. I can remove the X11 header if we want to be clear about it.

Comment (kkartaltepe) [reply plugins/obs-qsv11/CMakeLists.txt:61]: Sorry poked the libraries a bit and yea we should link it, ill add it in a bit.

File: plugins/obs-qsv11/common_utils_linux.cpp [hunk 2]
```diff
@@ -102,9 +107,29 @@ mfxStatus Initialize(mfxVersion ver, mfxSession *pSession,
...
+			vaGetDisplay((Display *)obs_get_nix_platform_display());
+	} else if (obs_get_nix_platform() == OBS_NIX_PLATFORM_WAYLAND) {
+		vaDisplay = vaGetDisplayWl(
+			(wl_display *)obs_get_nix_platform_display());
+	}
+
 	sts = MFXCreateSession(loader, 0, pSession);
 	MSDK_CHECK_RESULT(sts, MFX_ERR_NONE, sts);
 
+	// VPL Expects the VADisplay to be initialized.
```
Comment (RytoEX) [inline plugins/obs-qsv11/common_utils_linux.cpp:122]:
```suggestion
	// VPL expects the VADisplay to be initialized.
```

**Review by RytoEX — CHANGES_REQUESTED** [review]
One comment nit.

Awaiting other changes.

File: plugins/obs-qsv11/common_utils_linux.cpp (hunk 2 above)
Comment (kkartaltepe) [reply plugins/obs-qsv11/common_utils_linux.cpp:122]: Done.

**Review by RytoEX — APPROVED** [review]
Looks okay at a glance.

**Pull request merged**